core/src/test/java/org/apache/calcite/test/SqlValidatorTest.java

```diff
@@ -5238,6 +5238,35 @@ private ImmutableList<ImmutableBitSet> cube(ImmutableBitSet... sets) {
         .fails("Type mismatch in column 1 of UNION");
 
     sql("select ^slacker^ from emp union select name from dept").ok();
+
+    sql("select ^name^ from dept union select name from dept union select slacker from emp")
+        .withTypeCoercion(false)
+        .fails("Type mismatch in column 1 of UNION");
```

Comment:
Will the newly added case containing two unions have different behavior for this PR than the original case containing only one union? Do we need to add two additional union cases?

Reply:
yes, because then one of the union will be between `SELECT` and the result of the `UNION` or other operator.
>Do we need to add two additional union cases?

not sure about this since it doesn't introduce any new behavior